qemu/tests/block_hotplug_with_cpu_hotplug.py

```diff
@@ -0,0 +1,117 @@
+from virttest.qemu_devices import qdevices
+from virttest.qemu_capabilities import Flags
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test hotplug of block devices after
+    hotplug a cpu device.
+
+    1) Boot up guest
+    2) Hotplug device and verify in qtree
+    3) Check hotplug devices in guest
+    4) Hotpulg a cpu device
+    5) Hotplug device again and verify in qtree.
+    6) Check hotplug devices in guest
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    def find_all_disks(session):
+        """
+        Find all disks in guest.
+
+        :param session: The guest session object.
+        :return: disks set in guest.
+        """
+        global all_disks
+        all_disks = set(session.cmd('wmic diskdrive get index').split()[1:])
+        return all_disks
+
+    def block_check_in_guest(session, action, disks_before_plug,
+                             expected_num, test):
+        """
+        Check hotplug/unplug disks in guest.
+
+        :param session: The guest session object.
+        :param action: hotplug/unplug.
+        :param disks_before_plug: disks set before hotplug/unplug.
+        :param expcted_num: expected disks number in guest.
+        :param test: QEMU test object.
+        :return: changed disks set in guest.
+        """
+        if not utils_misc.wait_for(lambda: len(disks_before_plug ^ find_all_disks(
+               session)) == expected_num, 100, step=1.5):
+            disks_info_win = ('wmic logicaldisk get drivetype,name,description '
+                              '& wmic diskdrive list brief /format:list')
+            disks_info = session.cmd(disks_info_win)
+            test.log.info("The details of disks:\n %s", disks_info)
+            test.fail("Failed to {0} devices from guest, need to {0}: {1}, "
+                      "actual {0}: {2}".format(action, expected_num,
+                                               len(disks_before_plug ^ all_disks)))
+        return disks_before_plug ^ all_disks
+
+    def block_hotplug(image_name):
+        """
+        Hotplug disks and verify it in qtree.
+
+        :param image_name: Image name of hotplug disk
+        :return: List of objects for hotplug disk.
+        """
+        image_params = params.object_params(image_name)
+        devs = vm.devices.images_define_by_params(image_name,
+                                                  image_params, 'disk')
+        for dev in devs:
+            ret = vm.devices.simple_hotplug(dev, vm.monitor)
+            if ret[1] is False:
+                test.fail("Failed to hotplug device '%s'."
+                          "Output:\n%s" % (dev, ret[0]))
+        dtype = qdevices.QBlockdevNode if vm.check_capability(
+            Flags.BLOCKDEV) else qdevices.QDrive
+        devs = [dev for dev in devs if not isinstance(dev, dtype)]
+        return devs
```

Comment:
Suggest using the interface of the provider block_devices_plug.py.
There are such functions to help us to implement the above.

Reply:
@YongxueHong Thanks, will update lit later